File: Frontend/src/app/models/viewComment.ts
import { ViewReply } from "./viewReply";

export class ViewComment {
    constructor(
        public id: number,
        public userId: string,
        public userName: string,
        public dateCreated: Date,
        public body: string,
        public postId: number,
        public replies: ViewReply[]
    ) {}
}

File: Backend/Entity_Layer/Dtos/ViewComment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Entity_Layer.Dtos
{
    public class ViewComment
    {
        public long Id { get; set; } // commentId
        public string UserId { get; set; }
        public string UserName { get; set; }
        public DateTime DateCreated { get; set; }
        public string Body { get; set; }
        public long PostId { get; set; }
        public List<ViewReply> Replies { get; set; }
    }
}
